orchestrator-salt/src/main/resources/salt/salt/kerberos/config/cm-krb.j2

```diff
@@ -3,4 +3,7 @@ setsettings KRB_AUTH_ENABLE true
 setsettings SECURITY_REALM {{ salt['pillar.get']('kerberos:realm')|upper }}
 setsettings KDC_HOST {{ salt['pillar.get']('kerberos:url') }}
 setsettings KDC_ADMIN_HOST {{ salt['pillar.get']('kerberos:adminUrl') }}
-setsettings KDC_TYPE "Red Hat IPA"
\ No newline at end of file
+setsettings KDC_TYPE "Red Hat IPA"
```

Comment:
does it work with mixed deployments?
- old clusters e.g sdx have been deployed with old settings
- new clusters e.g new DHs are deployed with the new settings

Reply:
I haven't tested yet. I will do that, although I'm not sure how to confirm if it works or not.

Reply:
I launched a Data Lake with old encryption `rc4-hmac`. Keytab for Ranger:
```
Keytab name: FILE:/var/run/cloudera-scm-agent/process/1546333866-ranger-RANGER_ADMIN/ranger.keytab
KVNO Timestamp           Principal
---- ------------------- ------------------------------------------------------
   1 06/03/2020 05:26:38 HTTP/krisz-datalake-1-master0.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (aes256-cts-hmac-sha1-96)
   1 06/03/2020 05:26:38 HTTP/krisz-datalake-1-master0.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (aes128-cts-hmac-sha1-96)
   1 06/03/2020 05:26:38 HTTP/krisz-datalake-1-master0.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (aes256-cts-hmac-sha384-192)
   1 06/03/2020 05:26:38 HTTP/krisz-datalake-1-master0.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (aes128-cts-hmac-sha256-128)
   1 06/03/2020 05:26:38 HTTP/krisz-datalake-1-master0.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (des3-cbc-sha1)
   1 06/03/2020 05:26:38 HTTP/krisz-datalake-1-master0.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (arcfour-hmac)
   1 06/03/2020 05:26:38 rangeradmin/krisz-datalake-1-master0.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (aes256-cts-hmac-sha1-96)
   1 06/03/2020 05:26:38 rangeradmin/krisz-datalake-1-master0.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (aes128-cts-hmac-sha1-96)
   1 06/03/2020 05:26:38 rangeradmin/krisz-datalake-1-master0.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (aes256-cts-hmac-sha384-192)
   1 06/03/2020 05:26:38 rangeradmin/krisz-datalake-1-master0.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (aes128-cts-hmac-sha256-128)
   1 06/03/2020 05:26:38 rangeradmin/krisz-datalake-1-master0.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (des3-cbc-sha1)
   1 06/03/2020 05:26:38 rangeradmin/krisz-datalake-1-master0.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (arcfour-hmac)
```
Data Hub with encryption `aes256-cts`. HDFS keytab
```
Keytab name: FILE:/var/run/cloudera-scm-agent/process/1546334240-hdfs-NAMENODE/hdfs.keytab
KVNO Timestamp           Principal
---- ------------------- ------------------------------------------------------
   1 06/03/2020 06:00:49 HTTP/krisz-de-1-master3.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (aes256-cts-hmac-sha1-96)
   1 06/03/2020 06:00:49 HTTP/krisz-de-1-master3.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (aes128-cts-hmac-sha1-96)
   1 06/03/2020 06:00:49 HTTP/krisz-de-1-master3.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (aes256-cts-hmac-sha384-192)
   1 06/03/2020 06:00:49 HTTP/krisz-de-1-master3.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (aes128-cts-hmac-sha256-128)
   1 06/03/2020 06:00:49 HTTP/krisz-de-1-master3.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (des3-cbc-sha1)
   1 06/03/2020 06:00:49 HTTP/krisz-de-1-master3.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (arcfour-hmac)
   1 06/03/2020 06:00:49 hdfs/krisz-de-1-master3.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (aes256-cts-hmac-sha1-96)
   1 06/03/2020 06:00:49 hdfs/krisz-de-1-master3.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (aes128-cts-hmac-sha1-96)
   1 06/03/2020 06:00:49 hdfs/krisz-de-1-master3.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (aes256-cts-hmac-sha384-192)
   1 06/03/2020 06:00:49 hdfs/krisz-de-1-master3.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (aes128-cts-hmac-sha256-128)
   1 06/03/2020 06:00:49 hdfs/krisz-de-1-master3.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (des3-cbc-sha1)
   1 06/03/2020 06:00:49 hdfs/krisz-de-1-master3.wl.cloudera.site@KRISZ-EN.XCU2-8Y8X.WL.CLOUDERA.SITE (arcfour-hmac)
```
It seems that these keytabs use the same encryptions. Also tested HDFS access from Data Hub with env admin user and normal user. Env admin can access the cloud storage, but regular user cannot. Did the same with Hive where env admin user was able to create external table to S3
```
CREATE EXTERNAL TABLE `inventory`(
  `inv_item_sk` int,
  `inv_warehouse_sk` int,
  `inv_quantity_on_hand` int)
  PARTITIONED BY (
  `inv_date_sk` int) STORED AS ORC
  LOCATION
  's3a://cb-group/khorvath/tpcds_bin_partitioned2_orc_200.db/inventory';
```
while normal user was not
```
Error: Error while compiling statement: FAILED: HiveAccessControlException Permission denied: user [jozsi] does not have [READ] privilege on [s3a://cb-group/khorvath/tpcds_bin_partitioned3_orc_200.db/inventory] (state=42000,code=40000)
```